app/src/main/java/com/example/androidtraining/MainActivity.kt

```diff
@@ -37,10 +84,141 @@ fun Greeting(name: String, modifier: Modifier = Modifier) {
+        OutlinedTextField(
+            value = email.value,
+            onValueChange = { newValue ->
+                email.value = newValue
+            },
+            isError = !isValidEmail(email.value),
+            label = { Text("Email") },
+            placeholder = { Text("Input email") },
+            modifier = Modifier
+                .fillMaxWidth()
+        )
+        OutlinedTextField(value = password.value,
+            onValueChange = { password.value = it },
+            label = { Text("Password") },
+            singleLine = true,
+            placeholder = { Text("Password") },
+            modifier = Modifier
+                .fillMaxWidth(),
+            visualTransformation = if (passwordVisible.value) VisualTransformation.None else PasswordVisualTransformation(),
+            keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Password),
+            trailingIcon = {
+                val image = if (passwordVisible.value)
+                    painterResource(R.drawable.visibility)
+                else painterResource(R.drawable.visibility_off)
+
+                val description = if (passwordVisible.value) "Hide password" else "Show password"
+
+                IconButton(onClick = { passwordVisible.value = !passwordVisible.value }) {
+                    Image(painter = image, description)
+                }
+            })
+        Button(modifier = Modifier
+            .fillMaxWidth()
+            .padding(vertical = 5.dp, horizontal = 10.dp),
+            onClick = {
+                if (checkForValidCredentials(
+                        email.value,
+                        password.value
+                    )
+                ) navigateToProductScreen()
+                else errorVisible.value = true
+            }
+        ) {
+            Text(text = "Log in")
+        }
+        Text(text = if (errorVisible.value) "Invalid credentials" else "")
+    }
+}
+
+fun checkForValidCredentials(email: String, password: String): Boolean {
+    var validEmail = "marti@gmail.com"
+    var validPassword = "Pass123"
+    if (email == validEmail && password == validPassword) {
+        return true
+    }
+    return false
+}
+
+@Composable
+fun ProductScreen() {
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(horizontal = 20.dp, vertical = 20.dp)
+    ) {
+        Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
+            Icon(Icons.Default.ArrowBack, contentDescription = null)
+            Text(text = "Item")
+            Icon(Icons.Default.ShoppingCart, contentDescription = null)
+
+        }
+        Row(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(vertical = 20.dp),
+            horizontalArrangement = Arrangement.SpaceBetween
+        ) {
+            Image(painter = painterResource(id = R.drawable.photo), contentDescription = "tea pot")
```

Comment:
There's no need to wrap a single element -- you can set the modifiers to the `Image` itself. Were there issues with that approach so you wrapped it?

Reply:
@mtsoneva This comment seems to not be addressed. Should I merge the PR regardless?

Reply:
It is fixed. Maybe I forgot to mark it as resolved